CombinedResourceManager.cs

```diff
@@ -105,9 +103,21 @@ public override IList<ResourceRequiredContext> BuildRequiredResources(string str
 
                 IList<ResourceRequiredContext> result;
 
-                if (resourceType == ResourceType.Style) result = _combinatorService.CombineStylesheets(resources, settings);
-                else if (resourceType == ResourceType.JavaScript) result = _combinatorService.CombineScripts(resources, settings);
-                else return base.BuildRequiredResources(stringResourceType);
+                using (var mutex = new Mutex(initiallyOwned: false, "Global\\CombinatorFileAccess"))

```

Comment:
Functionally, this is the same as a simple `lock`. However, a `Mutex` is a heavy way of synchronization across processed, what you don't need. This can either be a simple [semaphore](https://learn.microsoft.com/en-us/dotnet/api/system.threading.semaphoreslim) as I mentioned (with any number of readers and a single writer) or you can do a slightly more sophisticated way with a named locking approach, so the locks are different per resource bundle.

Please decide with a team whether essentially making bundle writing serially executed (across the whole app) is acceptable, or you need a named lock.